deployer/src/persistence/mod.rs

```diff
@@ -691,7 +689,11 @@ impl StateRecorder for Persistence {
     type Err = Error;
 
     fn record_state(&self, state: DeploymentState) -> Result<()> {
-        self.state_send.send(state).map_err(Error::from)
+        let s = self.state_send.clone();
+        // move out of async context to be able to send
+        std::thread::spawn(move || s.blocking_send(state).map_err(Error::from))
```

Comment:
why do we need a thread and call `join` and not call `blocking_send` directly?

Reply:
A blocking send can't be done in a tokio thread, so that's why it needs to move out to a std thread. A `tokio::task::spawn_blocking` did not work.

Reply:
Solution: use unbounded channel for instant sends https://docs.rs/tokio/latest/tokio/sync/mpsc/index.html#communicating-between-sync-and-async-code